Pcap++/src/RawSocketDevice.cpp

```diff
@@ -505,12 +505,15 @@ namespace pcpp
 		}
 
 		// bind raw socket to interface
-		struct ifreq ifr;
-		memset(&ifr, 0, sizeof(ifr));
-		snprintf(ifr.ifr_name, sizeof(ifr.ifr_name), "%s", ifaceName.c_str());
-		if (setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, (void*)&ifr, sizeof(ifr)) == -1)
+		struct sockaddr_ll saddr;
+		memset(&saddr, 0, sizeof(saddr));
+		saddr.sll_family = AF_PACKET;
+		saddr.sll_protocol = htons(ETH_P_ALL);
+		saddr.sll_ifindex = if_nametoindex(ifaceName.c_str());
+
+		if (bind(fd, reinterpret_cast<struct sockaddr*>(&saddr), sizeof(saddr)) < 0)
```

Comment:
nit: Do we need the `struct` keyword here and on L508?

Reply:
Fixed in https://github.com/seladb/PcapPlusPlus/pull/1646/commits/c43e11019c27c046629d01382c59b4a70a4c778e